File: db/src/main/java/com/radioteria/db/entities/Track.java
package com.radioteria.db.entities;


import javax.persistence.*;

@Entity
@Table(name = "tracks")
public class Track extends BaseEntity<Long> implements Comparable<Track> {

    final public static String ID = "id";
    final public static String ORDER_ID = "order_id";
    final public static String TITLE = "title";
    final public static String ARTIST = "artist";
    final public static String DURATION = "duration";
    final public static String TRACK_FILE_ID = "track_file_id";
    final public static String CHANNEL_ID = "channel_id";

    @Id
    @GeneratedValue
    @Column(name = ID, nullable = false, updatable = false)
    private Long id;

    @Column(name = ORDER_ID, nullable = false)
    private Integer orderId;

    @Column(name = TITLE, nullable = false)
    private String title;

    @Column(name = ARTIST, nullable = false)
    private String artist;

    @Column(name = DURATION, nullable = false)
    private Long duration;

    @ManyToOne(targetEntity = File.class)
    @JoinColumn(name = TRACK_FILE_ID)
    private File trackFile;

    @ManyToOne(targetEntity = Channel.class)
    @JoinColumn(name = CHANNEL_ID, nullable = false)
    private Channel channel;

    public Track() {}

    public Track(String title, long duration) {
        this.setTitle(title);
        this.setDuration(duration);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Integer getOrderId() {
        return orderId;
    }

    public void setOrderId(Integer orderId) {
        this.orderId = orderId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getArtist() {
        return artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public Long getDuration() {
        return duration;
    }

    public void setDuration(Long duration) {
        this.duration = duration;
    }

    public File getTrackFile() {
        return trackFile;
    }

    public void setTrackFile(File trackFile) {
        this.trackFile = trackFile;
    }

    public Channel getChannel() {
        return channel;
    }

    public void setChannel(Channel channel) {
        this.channel = channel;
    }

    @Override
    public int compareTo(Track o) {
        return this.getOrderId() - o.getOrderId();
    }

    @Override
    public String toString() {
        return "Track{" +
                "id=" + id +
                ", orderId=" + orderId +
                ", title='" + title + '\'' +
                ", duration=" + duration +
                '}';
    }
}

File: db/src/main/kotlin/com/radioteria/db/entities/Track.kt
package com.radioteria.db.entities

import javax.persistence.*

object TrackMeta {
    const val TABLE_NAME = "tracks"
    const val TITLE = "title"
    const val ARTIST = "artist"
    const val DURATION = "duration"
    const val TRACK_FILE_ID = "track_file_id"
    const val CHANNEL_ID = "channel_id"
    const val ORDER_ID = "order_id"
}

@Entity
@Table(name = TrackMeta.TABLE_NAME)
class Track(
        @ManyToOne(targetEntity = File::class)
        @JoinColumn(name = TrackMeta.TRACK_FILE_ID, nullable = false)
        val trackFile: File,

        @ManyToOne(targetEntity = Channel::class)
        @JoinColumn(name = TrackMeta.CHANNEL_ID, nullable = false)
        val channel: Channel,

        @Column(name = TrackMeta.TITLE, nullable = false)
        val title: String = "",

        @Column(name = TrackMeta.ARTIST, nullable = false)
        val artist: String = "",

        @Column(name = TrackMeta.DURATION, nullable = false)
        val duration: Long = 0,

        @Column(name = TrackMeta.ORDER_ID, nullable = false)
        val orderId: Int = 0,

        id: Long? = null
) : IdAwareEntity<Long>(id)